bitcoin/src/blockdata/weight.rs

```diff
@@ -132,28 +208,31 @@ mod tests {
 
     #[test]
     fn kilo_weight_constructor() {
-        assert_eq!(Weight(1_000), Weight::from_kwu(1).expect("expected weight unit"));
+        assert_eq!(
+            Weight(1_000),
+            Weight::from_kwu(1).expect("weight unit from kwu does not overflow")
+        );
     }
 
     #[test]
     #[should_panic]
     fn kilo_weight_constructor_panic() {
-        Weight::from_kwu(u64::MAX).expect("expected weight unit");
+        Weight::from_kwu(u64::MAX).expect("weight unit from kwu does not overflow");
```

Comment:
This is factually incorrect.

Reply:
I'd prefer making this a normal test and just `assert!(Weight::from_kwu(u64::MAX).is_none())`. Conveys the meaning better.

Reply:
@tcharding I don't think this is factually incorrect.  If  there is a panic, it's because `from_kwu` produced an overflow.  The expect message is that we expect that not to happen which is consistent with the docs recommended expect message.  Of course, int this test I specifically make it overflow to test the panic case, so I can see how the message here might be misleading.

Reply:
@Kixunil your suggestion `assert!(Weight::from_kwu(u64::MAX).is_none())` does not panic.  This test is testing that `from_kwu` panics as expected.  Why do you recommend testing for `none`?

Reply:
Runn the current code and you'll see it always panics - that's the point!

Yes, my suggestion must not panic, that's also the point - remove the `#[should_panic]` attribute. `from_kwu` doesn't panic, it returns `Option`.

Reply:
Doh, I see now.  I've fixed this test here https://github.com/rust-bitcoin/rust-bitcoin/pull/2215/commits/a2c1183c8093f1091a328e5db052e6015c016597